sqlmesh/core/loader.py

```diff
@@ -186,11 +216,19 @@ def _load_sql_models(
                         dialect=config.model_defaults.dialect,
                         time_column_format=config.time_column_format,
                     )
-                    models[model.name] = model
 
-                    if isinstance(model, SeedModel):
-                        seed_path = model.seed_path
-                        self._track_file(seed_path)
+                cache_entry_name = "__".join(path.relative_to(models_path).parts).replace(
+                    ".sql", ""
+                )
+                cache_entry_id = self._model_cache_entry_id(path, context_path)
```

Comment:
What do you think about encapsulating name and id into get_or_load, so the cache stuff is less exposed?

Reply:
Encapsulate how exactly? Please note that `name` is a key of the entry while `entry_id` is used to invalidate the entry

Reply:
By passing in path and computing name and id  in get_or_load. model_cache init() would take in model_paths and context_path.